src/lib/topology-priv.h

```diff
@@ -0,0 +1,301 @@
+/*
```

Comment:
@singalsu do we need to duplicate this header file for the testbench and the firmware?

Reply:
I was thinking to eventually remove the parser from test bench and use this instead. But keep in test bench e.g. the library load code and other specific for it that this parser would call when compiled for test bench.